include/aws/iot/MqttRequestResponseClient.h

```diff
@@ -0,0 +1,378 @@
+#pragma once
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/crt/Exports.h>
+
+#include <aws/crt/Allocator.h>
+#include <aws/crt/Optional.h>
+#include <aws/crt/Types.h>
+#include <aws/crt/Variant.h>
+#include <aws/mqtt/request-response/request_response_client.h>
+
+#include <functional>
+
+namespace Aws
+{
+
+    namespace Crt
+    {
+        namespace Mqtt
+        {
+            class MqttConnection;
+        }
+
+        namespace Mqtt5
+        {
+            class Mqtt5Client;
+        }
+    } // namespace Crt
+
+    namespace Iot
+    {
+        namespace RequestResponse
+        {
+            /**
+             * The type of change to the state of a streaming operation subscription
+             */
+            enum class SubscriptionStatusEventType
+            {
+
+                /**
+                 * The streaming operation is successfully subscribed to its topic (filter)
+                 */
+                SubscriptionEstablished = ARRSSET_SUBSCRIPTION_ESTABLISHED,
+
+                /**
+                 * The streaming operation has temporarily lost its subscription to its topic (filter)
+                 */
+                SubscriptionLost = ARRSSET_SUBSCRIPTION_LOST,
+
+                /**
+                 * The streaming operation has entered a terminal state where it has given up trying to subscribe
+                 * to its topic (filter).  This is always due to user error (bad topic filter or IoT Core permission
+                 * policy).
+                 */
+                SubscriptionHalted = ARRSSET_SUBSCRIPTION_HALTED,
+            };
+
+            /**
+             * An event that describes a change in subscription status for a streaming operation.
+             */
+            class AWS_CRT_CPP_API SubscriptionStatusEvent
+            {
+              public:
+                SubscriptionStatusEvent() : m_type(SubscriptionStatusEventType::SubscriptionEstablished), m_errorCode(0)
+                {
+                }
+                SubscriptionStatusEvent(const SubscriptionStatusEvent &rhs) = default;
+                SubscriptionStatusEvent(SubscriptionStatusEvent &&rhs) = default;
+                ~SubscriptionStatusEvent() = default;
+
+                SubscriptionStatusEvent &operator=(const SubscriptionStatusEvent &rhs) = default;
+                SubscriptionStatusEvent &operator=(SubscriptionStatusEvent &&rhs) = default;
```

Comment:
nit: These 5 special members will be generated by the compiler. A user-defined default constructor doesn't prevent generating them.